lib/Linter.coffee

```diff
@@ -0,0 +1,55 @@
+{Task} = require('atom')
+
+workers = new Set
+workerFile = require.resolve('./core.coffee')
+
+module.exports =
+class Linter
+
+  name: 'CoffeeLint'
+  grammarScopes: [
+    'source.coffee'
+    'source.litcoffee'
+    'source.coffee.jsx'
+    'source.coffee.angular'
+  ]
+  scope: 'file'
+  lintsOnChange: true
+
+  destroy: ->
+    worker?.terminate() for worker in workers
+
+  lint: (textEditor) ->
+    filePath = textEditor.getPath()
+    if filePath
+      source = textEditor.getText()
+
+      isLiterate = textEditor.getGrammar().scopeName is 'source.litcoffee'
+
+
+      transform = ({level, message, rule, lineNumber, context}) ->
+        message = "#{message}. #{context}" if context
+        message = "#{message}. (#{rule})";
+
+        # Calculate range to make the error whole line
+        # without the indentation at begining of line
+        indentLevel = textEditor.indentationForBufferRow(lineNumber - 1)
+
+        startCol = (textEditor.getTabLength() * indentLevel)
+        endCol = textEditor.getBuffer().lineLengthForRow(lineNumber - 1)
+
+        range = [[lineNumber - 1, startCol], [lineNumber - 1, endCol]]
```

Comment:
Let's just bring in `atom-linter` and use [`generateRange`](https://github.com/steelbrain/atom-linter/tree/master#api) here like almost every other provider. This function has a much smarter version of this logic in place 😉.